pkg/operator/reporting/templates_test.go

```diff
@@ -0,0 +1,525 @@
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": ds1.Name},
+				},
+			},
+			expectOutput: fmt.Sprintf("SELECT * FROM hive.default.%s", ds1.Status.TableRef.Name),
+		},
+		{
+			name: "invalid report query with invalid templating (missing Inputs field) returns error and the expected error message",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery: newTestReportQuery(ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs. |}"),
+			},
+			templateContext: TemplateContext{},
+			expectErr:       true,
+			expectErrMsg:    "error parsing query: template: reportQueryTemplate:1: unexpected <.> in operand",
+		},
+		{
+			name: "valid report query with valid templating but undefined data source table ref returns error and the expected error message",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(ds2, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds2},
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable("", testNamespace, testNamespace, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": ds2.Name},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("error executing template: template: reportQueryTemplate:1:17: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: %s tableRef is empty", ds2.Name),
+		},
+		{
+			name: "valid report query with valid templating but the presto table's table ref returns error and the expected error message",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable("", testNamespace, testNamespace, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": ds1.Name},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("error executing template: template: reportQueryTemplate:1:17: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: tableRef PrestoTable %s not found", ds1.Status.TableRef.Name),
+		},
+		{
+			name: "ReportDataSource dependencies was not found (due to empty ds.Name) results in error and the expected error message",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(ds1, "SELECT labels['namespace'], FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
```

Comment:
Any reason some use `SELECT labels['namespace']` and some use `SELECT *`?  Maybe all of it should be consistent and just be `SELECT *`?

Reply:
I thought I had changed all the instances. I'll update.